R/zzz.R

```diff
@@ -35,3 +55,82 @@
   }
   packageStartupMessage(config_message)
 }
+
+python_is_compatible <- function(python_path) {
+  MIN_PYTHON_VERSION = "3.6"
+
+  Sys.setenv(RETICULATE_PYTHON = python_path)
+  python_info <- reticulate::py_discover_config()
+  Sys.unsetenv("RETICULATE_PYTHON")
+
+  if (!is.null(python_info$libpython) &&
+      utils::compareVersion(python_info$version, MIN_PYTHON_VERSION) >= 0) {
+    return(TRUE)
+  }
+  return(FALSE)
+}
+
+find_compatible_python <- function() {
+  python_path <- NULL
+  config_info <- reticulate::py_discover_config()
+  for (found_python in config_info$python_versions) {
+    if (python_is_compatible(found_python)) {
+      python_path <- found_python
+      break
+    }
+  }
+
+  if (is.null(python_path)) {
+    packageStartupMessage("No compatible version of Python (>=3.6, libpython libraries) found.")
```

Comment:
A clearer explanation of the issue here may be helpful to a user.

Reply:
I provided some more detail in the README since I'm not sure how useful a more detailed message would be in the command-line output and/or error message, but I see your point. I'll add another detailed comment or two.

Reply:
My assumption is that people will not read the README, and will instead copy and paste code to try to get it running as quickly as possible. Under that assumption, messages like these are really valuable.